_includes/promos/10tvpromo.html

```diff
@@ -0,0 +1,106 @@
+            <p class="subtitle" style="display:inline">Submit Your Business</p>
+        </div>
+    </div>
+    <br />
+    <form name="promo" class="needs-validation"   novalidate accept-charset="UTF-8"> <!--accept-charset="UTF-8" action="https://formkeep.com/f/exampletoken" method="POST"-->
+        <div class = "row">
+            <div class="col-md-5">
+                <input type="text" required class="form-control promoField" name="name"  placeholder="Business Name">
+                <div class="invalid-feedback">
+                    Please type in the business name
+                  </div>
+                <input type="url" class="form-control promoField" name="website"  required placeholder="Website URL">
+                <div class="invalid-feedback">
+                    Please type in a valid website URL
+                  </div>
+            </div> 
+            <div class="col-md-5">
+                <input type="text" required name="promo_text"  class="form-control promoField"  id="promo_text"  maxlength="100" placeholder="Promotional Text">
+                <div class="invalid-feedback">
+                    Please type in a valid text
+                  </div>
+                    <span id="charLeft"  style="color:grey; padding-left:10px; font-size:14px;"><span id="rem_post" title="1000">0</span>/100 characters   </span>
+                
+            </div>
+        </div>
+        <div class = "row">
+            <br />
+        </div>
+        <div class = "row">
+            <div class="col-md-12 text-center align-center">
+                 <button class="btn btn-primary dark-purple"  style="width:25%;margin-bottom:10px;" type="submit" id="submitButton">Submit</button>
+            </div>
+        </div>
+      </form>
+    <div class = "row">
+        <div class="col-md-12 text-center">
+            <span id="submitText" class="submitText">Thank you! Your ad will be up within 24 hours.</span>
+        </div>
+    </div>
+    <br>
+  </div>
+  
+
+  <script>
+    var copyText = document.getElementById("description");
+    var charLeft = document.getElementById("rem_post");
+    var submitText = document.getElementById("submitText");
+    var submitButton = document.getElementById("submitButton");
+    var forms = document.getElementsByClassName('needs-validation');
+   // 
+
+     copyText.addEventListener("input", function(){
+          var maxlength = this.getAttribute("maxlength");
+          var currentLength = this.value.length;
+
+          charLeft.innerText = currentLength;
+      });
+
+            (function() {
+            'use strict';
+            window.addEventListener('load', function() {
+                // Fetch all the forms we want to apply custom Bootstrap validation styles to
+                var forms = document.getElementsByClassName('needs-validation');
+                // Loop over them and prevent submission
+                var validation = Array.prototype.filter.call(forms, function(form) {
```

Comment:
Non-blocking but I'm curious over `Array.prototype.filter.call` vs `forms.filter` are you expecting a non array item to be within forms?

Reply:
This was just something from my collective research but you are right since we are just getting one form. I can change it whenever I can!

Reply:
Now that I think about it. I don't think we can filter a collection using `array.filter`.

We're getting forms via `document.getElementsByClassName('needs-validation');` which returns an HTMLCollection which is an array like but not an array. So I don't think we can call `forms.filter()`.

So what you're doing is good but I'm confused as to why we're filtering.